### PR TITLE
Added support for re-authenticating

### PR DESCRIPTION
**Pull request by anero**
Calling `authenticate` multiple times should not error, otherwise it's not possible to create a new session for same client if previous one expires.

### CODE REVIEW

**Comment by anero** [comment]
@javierjulio if interested also have updates available for upgrading `minitest` and `webmock`, let me know and I'll open a separated PR

**Comment by javierjulio** [comment]
@anero thanks for submitting! I'll get this in once the tests pass.

**Comment by javierjulio** [comment]
@anero the build failed but would you also be willing to update the gemspec with a minimum Ruby version? And also the Travis file? We should support only Ruby 2.3+ and Rails 4+. If we do that I can cut a new major release with those changes. Would you be willing to do that?

**Comment by anero** [comment]
Listo! (soy Argentino también :) )

**Comment by javierjulio** [comment]
@anero gracias! 😄 Estas en Buenos Aires? Hace tanto tiempo que no estoy alla!

Casi estamos listo aqui. Pienso que solamente tenemos que corregir otros dev dependencias. Voy hacer un review en que pienso debemos cambiar.

File: salesforcebulk.gemspec [hunk 1]
```diff
@@ -12,7 +12,9 @@ Gem::Specification.new do |s|
 
   s.files         = `git ls-files lib README.md`.split("\n")
 
-  s.add_dependency "activesupport", '>= 3.2.0', '< 6'
+  s.required_ruby_version = '>= 2.3'
+
+  s.add_dependency "activesupport", '>= 4', '< 6'
   s.add_dependency "xml-simple"
```
Comment (javierjulio) [inline salesforcebulk.gemspec]: Abajo de este linea, puedes cambiar:
- `shoulda` para que usa: `~> 3.5`
- `webmock` para que usa: `~> 3.0`

**Comment by javierjulio** [comment]
@anero la verdad no vale tener el Rails 4.0 gemfile para Travis porque ese version no es mas soportado. Por favor lo puedes sacar?

**Comment by javierjulio** [comment]
@anero awesome, build is passing so we are good!

**Comment by anero** [comment]
neat, thanks for the quick turnaround!

**Comment by javierjulio** [comment]
@anero no thank you! I just released 3.0 for you. I was going to ask you to make more edits but felt bad going back and forth so I did them. It amazes me after all these years I still get PRs for this. I don't use it but am very grateful for the PR as I'm sure it helps others. Thanks again! 🙏🏻😄

**Pull request merged**